packages/mui-material/src/Button/Button.js

```diff
@@ -398,22 +571,37 @@ const Button = React.forwardRef(function Button(inProps, ref) {
     </ButtonEndIcon>
   );
 
+  const loader = (
+    <ButtonLoadingIndicator className={classes.loadingIndicator} ownerState={ownerState}>
+      {loadingIndicator}
+    </ButtonLoadingIndicator>
+  );
+
   const positionClassName = buttonGroupButtonContextPositionClassName || '';
 
   return (
     <ButtonRoot
       ownerState={ownerState}
       className={clsx(contextProps.className, classes.root, className, positionClassName)}
       component={component}
-      disabled={disabled}
+      disabled={disabled || loading}
       focusRipple={!disableFocusRipple}
       focusVisibleClassName={clsx(classes.focusVisible, focusVisibleClassName)}
       ref={ref}
       type={type}
+      id={id}
       {...other}
       classes={classes}
     >
       {startIcon}
+      {enableLoadingWrapper ? (
+        // use plain HTML span to minimize the runtime overhead
+        <span className={classes.loadingWrapper} style={{ display: 'contents' }}>
```

Comment:
Can we add the style in the Button's styles using the class selector for this element? That way it's going to be easier to be overridden.

Reply:
Done

Reply:
Should we move `style={{ display: 'contents' }}` as well?